packages/flutter/lib/src/widgets/restoration_properties.dart

```diff
@@ -17,6 +17,78 @@ import 'restoration.dart';
 /// call [notifyListeners] from this method if the new value changes what
 /// [toPrimitives] returns.
 ///
+/// ## Using a RestorableValue
+///
+/// {@tool dartpad --template=stateful_widget_restoration}
+/// A [StatefulWidget] that has a restorable [int] property.
+///
+/// ```dart
+///   // The current value of the answer is stored in a [RestorableProperty].
+///   // During state restoration it is automatically restored to its old value.
+///   // If no restoration data is available to restore the answer from, it is
+///   // initialized to the specified default value, in this case 42.
+///   RestorableInt _answer = RestorableInt(42);
+///
+///   @override
+///   void restoreState(RestorationBucket oldBucket) {
+///     // All restorable properties must be registered with the mixin. After
+///     // registration, the answer either has its old value restored or is
+///     // initialized to its default value.
+///     registerForRestoration(_answer, 'answer');
+///   }
+///
+///   void _incrementAnswer() {
+///     setState(() {
+///       // The current value of the property can be accessed and modified via
+///       // the value getter and setter.
+///       _answer.value++;
+///     });
+///   }
+///
```

Comment:
For completeness, the RestorableInt needs to be disposed in `dispose`. See also https://github.com/flutter/flutter/pull/63497.